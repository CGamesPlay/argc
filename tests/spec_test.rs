use insta::assert_snapshot;

#[test]
fn test_spec_help() {
    assert_argc!(include_str!("spec.sh"), &["spec", "-h"]);
}

#[test]
fn test_spec_cmd_prefered_help() {
    assert_argc!(include_str!("spec.sh"), &["spec", "cmd-prefered", "-h"]);
}

#[test]
fn test_spec_cmd_omitted_help() {
    assert_argc!(include_str!("spec.sh"), &["spec", "cmd-omitted", "-h"]);
}

#[test]
fn test_spec_cmd_option_names_help() {
    assert_argc!(include_str!("spec.sh"), &["spec", "cmd-option-names", "-h"]);
}

#[test]
fn test_spec_cmd_option_formats_help() {
    assert_argc!(
        include_str!("spec.sh"),
        &["spec", "cmd-option-formats", "-h"]
    );
}

#[test]
fn test_spec_cmd_option_quotes_help() {
    assert_argc!(
        include_str!("spec.sh"),
        &["spec", "cmd-option-quotes", "-h"]
    );
}

#[test]
fn test_spec_cmd_flag_formats_help() {
    assert_argc!(include_str!("spec.sh"), &["spec", "cmd-flag-formats", "-h"]);
}

#[test]
fn test_spec_cmd_positional_only_help() {
    assert_argc!(
        include_str!("spec.sh"),
        &["spec", "cmd-positional-only", "-h"]
    );
}

#[test]
fn test_spec_cmd_positional_requires_help() {
    assert_argc!(
        include_str!("spec.sh"),
        &["spec", "cmd-positional-requires", "-h"]
    );
}

#[test]
fn test_spec_cmd_prefered_exec() {
    assert_argc!(
        include_str!("spec.sh"),
        &["spec", "cmd-prefered", "-f", "-o", "A", "AB", "C D"]
    );
}

#[test]
fn test_spec_cmd_option_names_exec() {
    assert_argc!(
        include_str!("spec.sh"),
        &[
            "spec",
            "cmd-option-names",
            "--opt2",
            "value2",
            "--opt3",
            "value3_0,value3_1",
            "--opt4",
            "value4_0",
            "--opt4",
            "value4_1",
            "--opt6",
            "a",
        ]
    );
}
