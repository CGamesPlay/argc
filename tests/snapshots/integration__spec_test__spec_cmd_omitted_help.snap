---
source: tests/spec_test.rs
assertion_line: 23
expression: output

---
RUN
spec cmd-omitted -h

STDOUT
exit 1

STDERR
spec-cmd-omitted 
Omitted

USAGE:
    spec cmd-omitted [OPTIONS] [ARG-1]

ARGS:
    <ARG-1>    

OPTIONS:
        --flag1           
    -h, --help            Print help information
        --opt1 <OPT-1>    


